.github/workflows/deploy-extension-to-marketplace.yml

```diff
@@ -40,17 +40,66 @@ jobs:
+              aws-access-key-id: ${{ secrets.AWS_MP_ACCESS_KEY }}
+              aws-secret-access-key: ${{ secrets.AWS_MP_ACCESS_KEY_ID }}
+              aws-region: us-east-1
+
           - name: Build Docker image
-            run: docker build -t 709825985650.dkr.ecr.us-east-1.amazonaws.com/liquibase/liquibase/liquibase-pro:${{ steps.get_latest_tag.outputs.LATEST_TAG || inputs.docker_tag }} .
+            run: docker build -t ${{ secrets.AWS_MP_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:${{ steps.get_latest_tag.outputs.LATEST_TAG }} .
 
           - name: Push to Docker AWS MP
-            run: docker push 709825985650.dkr.ecr.us-east-1.amazonaws.com/liquibase/liquibase/liquibase-pro:${{ steps.get_latest_tag.outputs.LATEST_TAG || inputs.docker_tag }}
+            run: docker push ${{ secrets.AWS_MP_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:${{ steps.get_latest_tag.outputs.LATEST_TAG }}
 
           - name: Submit Request for AWS MP Listing
             env:
               ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
-              IMAGE_TAG: ${{ steps.get_latest_tag.outputs.LATEST_TAG || inputs.docker_tag }}
+              IMAGE_TAG: ${{ steps.get_latest_tag.outputs.LATEST_TAG }}
               PRODUCT_ID: ${{ secrets.PRODUCT_ID }}
+              AWS_MP_REGISTRY: ${{ secrets.AWS_MP_REGISTRY }}
             run: |
               chmod +x ./.github/utils/submit-aws-mp-listing.sh
               ./.github/utils/submit-aws-mp-listing.sh
+
+  dry-run:
+    if: github.event_name == 'workflow_dispatch'
+    runs-on:
+      ubuntu-latest
+    env:
+      AWS_REGION: us-east-1
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Login to ECR Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ secrets.AWS_MP_REGISTRY }}
+          username: ${{ secrets.AWS_MP_ACCESS_KEY }}
+          password: ${{ secrets.AWS_MP_ACCESS_KEY_ID }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_MP_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_MP_ACCESS_KEY_ID }}
+          aws-region: us-east-1
+
+      - name: Build Docker image
+        run: docker build -t ${{ secrets.AWS_MP_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:${{ github.event.inputs.image_tag }} .
+
+      - name: Push to Docker AWS MP
+        run: docker push ${{ secrets.AWS_MP_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:${{ github.event.inputs.image_tag }}
+        continue-on-error: true
+
+      - name: Submit New Version Request for AWS MP Listing
+        env:
+          ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
+          IMAGE_TAG: ${{ github.event.inputs.image_tag }}
+          PRODUCT_ID: ${{ secrets.PRODUCT_ID }}
+          AWS_MP_REGISTRY: ${{ secrets.AWS_MP_REGISTRY }}
+        run: |
+          chmod +x ./.github/utils/submit-aws-mp-listing.sh
+          ./.github/utils/submit-aws-mp-listing.sh
```

Comment:
Do we actually want to submit a new version request during dry run?

Reply:
yes, we do but under a dry run and I need to edit it to "private" listing and not public. That way QA can test it

Reply:
more comment about "Private" and "Public" listing : https://datical.atlassian.net/browse/DAT-18744?focusedCommentId=151342